RefrigeratorSaraFarkash/RefrigeratorSaraFarkash/CRefrigerator.cs

```diff
@@ -0,0 +1,334 @@
+        {
+
+            if (this.spaceLeftInRefrigerator() < item1.TakeSpace)
+                Console.WriteLine("there is no space!!!");
+            else
+            {
+                for (int i = 0; i < this.numberOfShelves; i++)
+                {
+                    if (this.shelfs[i].spaceLeftInShelfs() > item1.TakeSpace)
+                    {
+                        item1.IdShelf = this.shelfs[i].Id;
+                        this.shelfs[i].AddItem(item1);
+                        Console.WriteLine("the item is added!!!");
+                        break;
+                    }
+                }
+            }
+        }
+
+        //4
+        public CItem removeItemFromRef(int idItem)
+        {
+            CItem itemRemove;
+            for (int i = 0; i < this.numberOfShelves; i++)
+            {
+                if (shelfs[i].findItemInShelf(idItem))
+                {
+                    itemRemove = shelfs[i].removeItemfromShelf(idItem);
+                    return itemRemove;
+                }
+            }
+
+            Console.WriteLine("the item not found!!!");
+            return null;
+        }
+        //5
+        public void cleanExpired()
+        {
+            for (int i = 0; i < shelfs.Count; i++)
+            {
+                List<CItem> itemsToRemove = new List<CItem>();
+                foreach (CItem item in shelfs[i].Items)
+                {
+                    if (item.ExpiryDate < DateTime.Today)
+                    {
+                        Console.WriteLine("the product is" + item.ToString()
+                        + "Thrown in the trash due to expiration");
+                        itemsToRemove.Add(item);
+                    }
+                }
+                if (itemsToRemove.Count >= 0)
+                {
+                    // Since it is impossible to delete while in a loop, I created a list that stores all the expired products.
+                    foreach (CItem item in itemsToRemove)
+                    {
+                        CItem itemremoving = removeItemFromRef(item.Id);
+                    }
+                }
+            }
+        }
+
+        //6
+        public List<CItem> wantEat(string kashroot, string typeeat)
+        {
+            List<CItem> wanttoeat = new List<CItem>();
+
+            for (int i = 0; i < shelfs.Count; i++)
+            {
+                foreach (CItem item in shelfs[i].Items)
+                {
+                    if (item.Kashroot == kashroot && item.TypeI == typeeat && item.ExpiryDate > DateTime.Today)
+                        wanttoeat.Add(item);
+                }
+            }
+            return wanttoeat;
+        }
+
+        // sort
+
+        public int CompareTo(CRefrigerator otherRefrigerator)
+        {
+            return ((this.spaceLeftInRefrigerator() > otherRefrigerator.spaceLeftInRefrigerator()) ? (-1) : (this.spaceLeftInRefrigerator() == otherRefrigerator.spaceLeftInRefrigerator()) ? 0 : 1);
```

Comment:
This can result in calling the functions twice and this code is unreadable.
Split the code to lines so you call the space functions only once as well as make the return line readable and short